src/App.tsx

```diff
@@ -15,36 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  BY_LENGTH = 'BY_LENGTH',
+  BY_ALPHABET = 'BY_ALPHABET',
+  DEFAULT = 'DEFAULT',
+}
+
+interface SortOptions {
+  sortField: SortType;
+  isReversed: boolean;
+}
+
+function preparedList(goods: string[], { sortField, isReversed }: SortOptions) {
+  let prepared = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SortType.BY_ALPHABET:
+        prepared.sort((item1, item2) => item1.localeCompare(item2));
+        break;
+      case SortType.BY_LENGTH:
+        prepared.sort((item1, item2) => item1.length - item2.length);
+        break;
+      case SortType.DEFAULT:
+        prepared = [...goods];
+        break;
+    }
+  }
+
+  if (isReversed) {
+    prepared.reverse();
+  }
+
+  return prepared;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.DEFAULT);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleList = preparedList(goodsFromServer, { sortField, isReversed });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField === SortType.BY_ALPHABET,
+          })}
```

Comment:
[UX] - It's good to mark the active button, but the current implementation may not be clear enough to the user. Consider adding additional visual cues or text to indicate more clearly which sorting option is currently active.